rfd/xxxx-tsh-bench-db.md

```diff
@@ -0,0 +1,166 @@
+
+#### Connection establishment
+It covers authorization capabilities and also initial connections. Although the
+other suites will also cover this, we can have more options, like providing
+different users and having the ability to run it across different database
+protocols. It will not rely on any table or struct on the connected database, as
+it will only perform a ping to ensure the connection is healthy. For databases
+without support to ping, a simple query will be sent. For example, the suite
+will send a `SELECT 1;` query on PostgreSQL databases.
+
+This suite can also validate changes on the access checker without running large
+amounts of queries.
+
+```code
+$ tsh bench db connect --db-user=postgres --db-name=postgres postgres-dev
+```
+
+#### Import/Export data
+It is a typical flow for users to import data to their databases. This suite is
+focused on running multiple queries within the same database connection (much
+like what an import tool would do). To do so, some databases will require an
+additional step executed before the suite starts. For example, on PostgreSQL,
+the necessary tables must be created. This setup won't be performed inside the
+workload function.
+
+The exporting suite will load data outside the workload function. This way, the
+measuring will only include read queries.
+
+##### Data generation
+The database clients will have access to a set of functions that can be used to
+generate random data. Those functions will cover the following types: `[]byte`,
+`string`,  `bool`, `int64`, and `float64`. Some of those functions, such as
+generating `[]byte` and `string`, will make it possible to define their length.
+The clients will be responsible for determining their structure (if required,
+creating them on the `Setup` call) and filling with data on `CreateRecord`.
+
+### Direct database connections
+To directly connect to the databases, the suites must accept the credentials
+that will be used. To avoid adding multiple flags to cover the scenarios, we're
+heading into having a single `--uri` flag, which will receive the database URI
+with all options included. This way, users can choose the authentication method
+that will be used.
+
+```code
+$ tsh bench db connect --uri postgres://hello@localhost:5432
+```
+
+By providing this flag, the `tsh` will connect directly to the database without
+using any Teleport certificate.
+
+## Security
+
+### Bypass MFA or access checker
+The command will use the same flow used by `tsh db login` to issue certificates,
+including support to MFA. The main difference is that the issue credentials
+won't be persisted on disk.
+
+### Malicious URI on the direct database connection
+Users may try to send malicious URI when connecting directly to databases.
+Teleport doesn't validate the input provided; it is passed directly to the
+database client responsible for validating it. Client execution is entirely
+done locally. Nothing is sent to the Teleport server.
+
+## Future work
+
```

Comment:
This would also be useful for reproducing bugs, as users could provide the recorded file. But it might require further investigation and sounds out of this RFD scope.

However, we might get the first version by [implementing custom queries and client execution](https://github.com/gravitational/teleport/pull/27856#discussion_r1230954799). What do you think?